### PR TITLE
[3548][ADD] product_delivery_destination: split from product_yahoo_auction_sst

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3548](https://www.quartile.co/web?#id=3548&action=771&model=project.task&view_type=form&menu_id=505)

### CODE REVIEW

File: product_yahoo_auction_sst/models/deliver_prefecture.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class DeliverPrefecture(models.Model):
+    _name = 'deliver.prefecture'
+    _description = 'Delivery Prefecture'
+
+    name = fields.Char(string='Prefecture Name', required=True)
```
Comment (yostashiro) [inline product_yahoo_auction_sst/models/deliver_prefecture.py]: Shouldn't we extend `res.country.state` instead by adding `is_deliverable` boolean field, perhaps?

File: product_yahoo_auction_sst/models/delivery_city.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class DeliveryCity(models.Model):
+    _name = 'delivery.city'
+    _description = 'Delivery Cities'
+
+    name = fields.Char(string='City Name', required=True)
```
Comment (yostashiro) [inline product_yahoo_auction_sst/models/delivery_city.py]: Add `state_id` with many2one relation to `res.country.state`.

**Review by yostashiro — COMMENTED** [review]
Partial review.

**Comment by yostashiro** [comment]
@kanda999 Shouldn't this actually be a separate module to product_yahoo_auction_sst?

**Comment by kanda999** [comment]
@AungKoKoLin1997 
I checked with the customer and they said that the view of list may be changed.
So please adjust this PR function as sperate module.

**Comment by AungKoKoLin1997** [comment]
@kanda999  Please review this PR and let me know if you have any better module name.

File: delivery_information/__manifest__.py
Comment (yostashiro) [inline delivery_information/__manifest__.py:1]: Please rename the module to `product_delivery_destination`.

@kanda999 Please make a counter-suggestion if you can think of a better name.

**Review by yostashiro — CHANGES_REQUESTED** [review]
Why mix words `deliver` and `delivery`?  Please be strict and consistent with word choices.

File: product_delivery_destination/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Product Delivery Destination",
+    "version": "11.0.1.0.0",
+    "author": "Quartile Limited",
+    "website": "https://www.quartile.co",
+    "category": "Product",
+    "license": "AGPL-3",
+    "depends": ["base", "product"],
```
Comment (yostashiro) [inline product_delivery_destination/__manifest__.py]:
```suggestion
    "depends": ["product"],
```

File: product_delivery_destination/models/product_template.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    delivery_prefecture = fields.Many2one('res.country.state', string="Delivery Prefecture", domain="[('is_deliverable','=', True)]")
+    delivery_city = fields.Many2one('delivery.city', string="Delivery City")
```
Comment (yostashiro) [inline product_delivery_destination/models/product_template.py]: To follow the standard naming rule.

```suggestion
    delivery_state_id = fields.Many2one('res.country.state', string="Delivery Prefecture", domain="[('is_deliverable','=', True)]")
    delivery_city_id = fields.Many2one('delivery.city', string="Delivery City")
```

File: product_delivery_destination/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1,2 @@
+This module introduces additional fields related to delivery specifics,
+associated with the state and city.
```
Comment (yostashiro) [inline product_delivery_destination/readme/DESCRIPTION.rst]:
```suggestion
This module adds state and city many2one fields to the product model.

These fields are expected to be used to keep the information necessary for listing products in an external auction/marketplace website.
```

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: product_delivery_destination/models/product_template.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    def _onchange_delivery_city_id(self):
+        if self.delivery_city_id:
+            self.delivery_state_id = self.delivery_city_id.state_id
+    
+    @api.onchange("delivery_state_id")
+    def _onchange_delivery_state_id(self):
+        if self.delivery_state_id:
+            return {'domain': {'delivery_city_id': [('state_id', '=', self.delivery_state_id.id)]}}
+        else:
+            return {'domain': {'delivery_city_id': []}}
```
Comment (yostashiro) [inline product_delivery_destination/models/product_template.py]: I missed to comment about this earlier. It's not a good practice to manipulate wiht the domain with an onchange method, as the domain doesn't stick when the view is reloaded.

We may consider to do something like this instead: https://github.com/OCA/bank-payment/pull/1106/commits/3955f46146919e60bb9d91ad91a1542c7796a835#diff-40552b82ea2aa604c7524ad4e88fb50e0afc7f780d70e3bafe4336444aec2c79R74-R87

**Review by yostashiro — COMMENTED** [review]
Please add CONFIGURE.rst as well with a short explanation.

**Review by yostashiro — APPROVED** [review]
Code review.

**Review by kanda999 — APPROVED** [review]
Functional review. It works as expected

**Pull request merged**